### ISSUE TITLE
Campaigns Index endpoint

### ISSUE THREAD

**Issue by mendelB** [DoSomething/phoenix-next#942]
### User Story
As a developer, I want to implement an index endpoint for `/campaigns` on Phoenix.

#### Additional Information
After some discussion, we've fine-tuned our need at the moment to adding support for batched ID requests to this endpoint. E.G.
`/campaigns?ids=1,2,3`

We'd limit this to a small number of IDs (in the 10-15 range), and merge a return value of the Contentful campaigns for all provided Contentful IDs and (transformed-to-align-schema) Ashes campaigns based on the provided ashes IDs.

Eventually, we'd like to implement a caching system using Fastly and a Contentful webhook so that services can be comfortable sending us multiple 'show' requests instead of the somewhat inelegant batched ID requests.

_Any other info or references to cards/tickets from other services (Trello, etc)._
Pivotal ID: [#157410894](https://www.pivotaltracker.com/story/show/157410894)
Pivotal ID: [#155672260](https://www.pivotaltracker.com/n/projects/2019429/stories/155672260)

### Why This Matters
This is another step towards a full-fledged Phoenix Campaigns API.
We'd also like to act as a proxy for the Ashes `/campaigns` index endpoint, to help do away with multiple services querying Ashes, and instead, in addition to us serving Contentful campaigns from this endpoint, we'd also be supporting Ashes campaigns.

### Tasks
This feature includes the following tasks:
- [ ] Add batched ID `/campaigns` endpoint support for Contentful and Ashes IDs

**Comment by mendelB** [comment]
https://dosomething.slack.com/archives/C3ASB4204/p1527193987000147
^ we discuss the value to query against for Contentful campaigns (`legacyCampaignId` vs. `sys.id`) and determine that the approach should be to query likely legacy IDs against Contentful, then filter the IDs that were not found, and pass those over to an Ashes query.

**Comment by weerd** [comment]
Side note, I think we want something like: `/campaigns?filter[id]=1,2,3`.

Rogue's docs provide some good [examples](https://github.com/DoSomething/rogue/blob/master/docs/endpoints/posts.md)!

**Comment by DFurnes** [comment]
Yep! We've settled on a pattern [suggested by JSON API](http://jsonapi.org/recommendations/#filtering), where we "namespace" filters under a `filter[]` query parameter. This way we wouldn't run into naming collisions one day if, for example, we wanted to filter something by pages (e.g. `?filter[page]=something&page=2`).

**Comment by mendelB** [comment]
Ah, that makes good sense! I'll update the description.

**Comment by weerd** [comment]
In regards to querying by IDs, copy-pasting from the slack convo:

------

So I think the idea was more using something similar to this:

```php
public function show($id)
{
    if (is_legacy_id($id)) {
        $data = $this->campaignRepository->findByLegacyCampaignId($id);
    } else {
        $data = $this->campaignRepository->getCampaign($id);
    }

    return response()->json(['data' => $data]);
}
```
where if legacy IDs are passed, we use `is_legacy_id()` to know that it _is likely_ a legacy ID, and if so _first_ we ask Contentful to find campaigns by `legacyCampaignId`, whatever it doesn’t find, we then query Ashes.

That’s basically how the `findByLegacyCampaignId()` method works when it’s called (after confirming it’s a legacy id passed).

@mendelB mentioned finding this in the Contentful API docs: https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/search-parameters/inclusion/query-entries/console/php

And that could likely work `->where('fields.legacyCampaignId', ['362, '118', '1280], 'in');`

We’ll definitely have to do a little processing to find which initially supplied IDs didn’t return anything via a few helper methods or function, and then send that batch to Ashes.
